File: benchmarks/langages1/count/n.cpp
#include <iostream>
#include <fstream>
#include <string>

int main() {
    std::ifstream file("langages1/count/n.txt");
    if (!file) {
        std::cerr << "❌ Erreur : Le fichier 'n.txt' est introuvable\n";
        return 1;
    }

    std::string line;
    if (!std::getline(file, line)) {
        std::cerr << "❌ Erreur : Impossible de lire le fichier 'n.txt'\n";
        return 1;
    }

    try {
        long long n = std::stoll(line) + 1;
        for (long long i = 1; i < n; ++i) {
            if (i < 4) {
                std::cout << i << ", ";
            }
        }
        std::cout << "\n";
    } catch (...) {
        std::cerr << "❌ Erreur : Le fichier 'n.txt' ne contient pas un entier valide\n";
        return 1;
    }

    return 0;
}

File: benchmarks/langages1/count/n.rs
use std::fs::read_to_string;
use std::path::Path;
// use std::num::ParseIntError;

fn main() {
    // Chemin du fichier dans le même dossier
    let path = Path::new("langages1/count/n.txt");

    if !path.exists() {
        eprintln!("❌ Erreur : Le fichier 'n.txt' est introuvable");
        return;
    }

    match read_to_string(path) {
        Ok(content) => match content.trim().parse::<u64>() {
            Ok(n) => {
                let n_plus = n + 1;
                for i in 1..n_plus {
                    if i < 4 {
                        print!("{}, ", i);
                    }
                }
                println!();
            }
            Err(_e) => {
                eprintln!("❌ Erreur : Le fichier 'n.txt' ne contient pas un entier valide");
            }
        },
        Err(_e) => {
            eprintln!("❌ Erreur : Impossible de lire le fichier 'n.txt'");
        }
    }
}
